lib/rubocop/cop/rspec/spec_file_path_format.rb

```diff
@@ -59,6 +65,53 @@ def on_top_level_example_group(node)
 
         private
 
+        # Inflector module that uses ActiveSupport for advanced inflection rules
+        module ActiveSupportInflector
+          def self.call(string)
+            ActiveSupport::Inflector.underscore(string)
+          end
+
+          def self.prepare_availability(config)
+            return if @prepared
+
+            @prepared = true
+
+            inflector_path = config.fetch('InflectorPath')
+
+            unless File.exist?(inflector_path)
+              raise "The configured `InflectorPath` #{inflector_path} does " \
+                    'not exist.'
+            end
+
+            require 'active_support/inflector'
+            require inflector_path
+          end
+        end
+
+        # Inflector module that uses basic regex-based conversion
+        module DefaultInflector
+          def self.call(string)
+            string
+              .gsub(/([^A-Z])([A-Z]+)/, '\1_\2')
+              .gsub(/([A-Z])([A-Z][^A-Z\d]+)/, '\1_\2')
+              .downcase
+          end
+        end
+
+        def inflector
+          case cop_config.fetch('EnforcedInflector')
+          when 'active_support'
+            ActiveSupportInflector.prepare_availability(cop_config)
```

Comment:
In addition to the excessive caching across directories, there can be another problem - multiple threads. RuboCop can run in threads? Or are those processes?

In any case, can we init the inflector once per inspection?

Or this would be too expensive if the inflection file is large?

I can think of a monorepo with many engines each with its own set of inflections